diff --git a/Code/ObjectMapping/RKObjectLoader.m b/Code/ObjectMapping/RKObjectLoader.m
index eed3107..51e2c07 100644
--- a/Code/ObjectMapping/RKObjectLoader.m
+++ b/Code/ObjectMapping/RKObjectLoader.m
@@ -292,13 +292,14 @@
     if ([self.response isServiceUnavailable]) {
         [[NSNotificationCenter defaultCenter] postNotificationName:RKServiceDidBecomeUnavailableNotification object:self];
     }
+    NSCharacterSet *set = [NSCharacterSet whitespaceCharacterSet];
 
     if ([self.response isFailure]) {
         [self informDelegateOfError:self.response.failureError];
 
         [self didFailLoadWithError:self.response.failureError];
         return NO;
-    } else if ([self.response isNoContent]) {
+    } else if ([[self.response.bodyAsString stringByTrimmingCharactersInSet: set] length] == 0 && [self.response isSuccessful]) {
         // The No Content (204) response will never have a message body or a MIME Type.
         id resultDictionary = nil;
         if (self.targetObject) {
